### DISCUSSION TITLE
Tonight my Ideavim broke with update 2.6.0...

### DISCUSSION THREAD

**Discussion by clarkdowding** [JetBrains/ideavim#735] [Q&A] [answered]
I was happily coding along in PyCharm Pro... enjoying using vim. 
Up pops a little notification that there is an update for Ideavim. I accept the update.
Instantly I can no longer:

(1) Exit from Insert to Normal mode with ESC. Have to use Control-C.
(2) When I try to use O or Shift-O to open a new line in Insert mode, the cursor stays on the current line and shifts to a broken insert mode. By that I mean the cursor changes from block to a slim insert cursor... but typing does not insert text. Have to hit Control-C.

Researched fixes for hours. No joy.
I'd sure like to rollback to whatever prior version of Ideavim I was on.

If these symptoms seem familiar to someone, I'd love to know some steps to try to fix this. I've tried a lot of things. 

Clark

**Comment by TwelveNight** [comment]
the same

**Comment by lubiksss** [comment]
I have same issue, this is very annoying me
Is there anyway downgrade idea vim version?

**Reply by lubiksss** [reply]
i found this https://plugins.jetbrains.com/plugin/164-ideavim/versions/stable
you can download previous version of ideavim and 2.5.1 version has no such issue

**Reply by clarkdowding** [reply]
Thank you @lubiksss. Rolling back indeed fixed the broken ```ESC``` to leave _insert mode_ and ```Shift-O``` to open new line in _insert mode_ released in version 2.6.0 last night.

I found [these instructions](https://www.jetbrains.com/help/pycharm/managing-plugins.html) for manually installing a plugin. That, together with @lubkiss solution, fixed the problem for me.

It should be noted that the devs released 2.6.1 early this morning and this bug is **still** present there.

SOLUTION:

## Roll back from 2.6.0 or 2.6.1 to version 2.5.1.

Thanks all.
Clark

**Comment by icodealot** [comment]
Same, 2.6.0 seemed to be bugged, so I rolled back to 2.5.1 manually.

On Mac, cmd+f stopped working for me, so I figured I'd just use `/<search>`, but even that got locked into visual select mode after jumping. (it seemed like `esc` no longer worked momentarily)

**Comment by jlandahl** [comment]
Same issue here. Really frustrating, but downgrading to 2.5.1 fixed it.

**Comment by AlexPl292** [comment, marked as answer]
Hi! The 2.6.0 and 2.6.1 versions of IdeaVim are broken. Please update to the latest ~~2.6.2~~ 2.6.3 version.

**Reply by Korolev-Oleg** [reply]
Hello, I updated IdeaVim now. When I switch from insert to command mode and try to execute `shift + o`, instead of inserting a new line above, it simply moves the cursor one line up. The same thing happens for `o`.

**Reply by AlexPl292** [reply]
Could you please attach the log (https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files) with that?

**Reply by Korolev-Oleg** [reply]
I sent it to [AlexPl292@gmail.com](mailto:AlexPl292@gmail.com) with title `logfile for idea vim discussion 735`

**Reply by Korolev-Oleg** [reply]
I also have the setting inoremap `jk <esc>` `in ~/.ideavimrc` and now, to enter command mode, I need to enter `jkjk` instead of `jk` as before.

**Reply by ka-sandor** [reply]
Installed 2.6.2, had the same problems. Downgrading helped.

**Reply by lpakula** [reply]
Same problem, 2.6.2, I can type letters and numbers but some of the keys are not working eg: esc, ctrl + a, backspace (ubuntu 22.04)

**Reply by kgish** [reply]
The ideavim plugin is completely broken after the last update, please fix!

**Reply by kgish** [reply]
How does one downgrade?

**Reply by AlexPl292** [reply]
Please try the uploaded 2.6.3 version.

**Reply by lpakula** [reply]
@AlexPl292 2.6.3 seems to be working ok, thanks!

**Reply by kgish** [reply]
For me, too!

**Reply by clarkdowding** [reply]
Thanks for getting it fixed @AlexPl292 :)

**Comment by AlexPl292** [comment]
Hi there! Releases 2.6.0, 2.6.1, and 2.6.2 have broken behavior. Release 2.6.3 should work well and it's based on the previous 2.5.1 release with several critical fixes. All changes mentioned in the changelog related to 2.6.0 are not included in 2.6.3.

I deeply apologize for any inconvenience caused by this set of broken releases. We faced several issues and obstacles, so we were not able to release the fixed version quickly enough. We'll run an internal review of this incident to avoid such issues in the future.

The version with the fixes/changes that were mentioned for 2.6.0 will be released in the next version of IdeaVim as soon as we deal with all reported issues. Thank you.

**Reply by samtron1412** [reply]
Hi @AlexPl292, thanks for the quick fix!
However, I still have some issues with having IdeaVim plugin both in host and client with CodeWithMe session. Do you know how to resolve the issues? I needed to disable/uninstall IdeaVim plugin on host IDE to allow for the client to use the plugin normally without issues.

**Reply by AlexPl292** [reply]
Hi, you should go to settings and remove the plugin from "Plugins (Host)". IdeaVim should work on the client side only.

**Reply by Korolev-Oleg** [reply]
@AlexPl292 Thank you for supporting the IdeaVim plugin; it's hard to imagine working with the IDE without
